src/object/obj_class.c

```diff
@@ -328,3 +356,127 @@ daos_oclass_grp_nr(struct daos_oclass_attr *oc_attr, struct daos_obj_md *md)
 	/* NB: @md is unsupported for now */
 	return oc_attr->ca_grp_nr;
 }
+
+/** a structure to map EC object class to gf tables */
+struct daos_oc_gftbls {
+	/** object class id */
+	daos_oclass_id_t	 e_oc_id;
+	/**
+	 * pointer to array of input tables generated from coding coefficients.
+	 */
+	unsigned char		*e_gftbls;
+};
+
+static struct daos_oc_gftbls	*daos_oc_gftbls_map;
+static int			 daos_oc_gftbls_nr;
+
+void
+daos_oc_gftbls_map_fini()
+{
+	struct daos_oc_gftbls	*oc_gftbls;
+	struct daos_obj_class	*oc;
+	int			 ocnr = 0;
+	int			 i;
+
+	if (daos_oc_gftbls_map == NULL)
+		return;
+
+	for (oc = &daos_obj_classes[0]; oc->oc_id != DAOS_OC_UNKNOWN; oc++) {
+		if (oc->oc_attr.ca_resil == DAOS_RES_EC)
+			ocnr++;
+	}
+	D_ASSERT(daos_oc_gftbls_nr == ocnr);
+
+	for (i = 0; i < ocnr; i++) {
+		oc_gftbls = &daos_oc_gftbls_map[i++];
+		if (oc_gftbls->e_gftbls != NULL)
+			D_FREE(oc_gftbls->e_gftbls);
+	}
+
+	D_FREE(daos_oc_gftbls_map);
+	daos_oc_gftbls_map = NULL;
+	daos_oc_gftbls_nr = 0;
+}
+
+int
+daos_oc_gftbls_map_init()
+{
+	struct daos_oc_gftbls	*oc_gftbls;
+	struct daos_obj_class	*oc;
+	unsigned char		*encode_matrix = NULL;
+	int			 ocnr = 0;
+	int			 i;
+	int			 k, p, m;
+	int			 rc;
+
+	if (daos_oc_gftbls_map != NULL)
+		return 0;
+
+	for (oc = &daos_obj_classes[0]; oc->oc_id != DAOS_OC_UNKNOWN; oc++) {
+		if (oc->oc_attr.ca_resil == DAOS_RES_EC)
+			ocnr++;
+	}
+
+	if (ocnr == 0)
+		return 0;
+
+	D_ALLOC_ARRAY(daos_oc_gftbls_map, ocnr);
+	if (daos_oc_gftbls_map == NULL)
+		D_GOTO(failed, rc = -DER_NOMEM);
+
+	i = 0;
+	for (oc = &daos_obj_classes[0]; oc->oc_id != DAOS_OC_UNKNOWN; oc++) {
+		if (oc->oc_attr.ca_resil != DAOS_RES_EC)
+			continue;
+
+		oc_gftbls = &daos_oc_gftbls_map[i++];
```

Comment:
it is 1:1 mapping between oc and the GF tables, but I think should not put it into obj class because:
1) obj class struct is user visible (user can define/add new obj class although not implemented yet), but the GF tables is EC internal handling related.
2) some other info need to associated with oc_id for example the encoding matrix (I added by refresh the patch). this is also only related internal EC implementation.

Reply:
ah, indeed, makes sense. Thanks.